imageregistry/v1/types.go

```diff
@@ -70,8 +72,9 @@ type ImageRegistrySpec struct {
 	Routes []ImageRegistryConfigRoute `json:"routes,omitempty" protobuf:"bytes,9,rep,name=routes"`
 	// replicas determines the number of registry instances to run.
 	Replicas int32 `json:"replicas" protobuf:"varint,10,opt,name=replicas"`
-	// logging determines the level of logging enabled in the registry.
-	LogLevel int64 `json:"logging" protobuf:"varint,11,opt,name=logging"`
+	// logging is deprecated, use logLevel instead.
+	// +optional
+	Logging *int64 `json:"logging,omitempty" protobuf:"varint,11,opt,name=logging"`
```

Comment:
AFAIK we cannot change `int64` -> `*int64`, but I'd like to get an input from @adambkaplan

Reply:
This depends on https://github.com/openshift/api/pull/640#discussion_r429934674

Reply:
This is interesting. In normal circumstances this would be a breaking change, but in our case it is not.

In the golang code we are introducing the new field `Logging`, so technically this is not a breaking (golang) API change.

For the JSON API, we are converting a field from required -> optional, which I believe is a non-breaking change. However, not setting anything implies a default of `0`, so we need to respect that in our migration code:

1. `logLevel = 0`, `logging = nil` -> logs should use `-v 0`
2. `logLevel = 0`, `logging = <non-nil>` -> logs should use `-v <non-nil>`.
3. `logLevel = <non-zero>` -> logs should use `-v <non-zero>`

Reply:
You can make this `omitempty`, but you can't change the type. @adambkaplan had similar discussion when working on https://github.com/openshift/api/pull/563 with @deads2k

Reply:
I brought back the `int64`. Migration path will look like https://github.com/openshift/cluster-image-registry-operator/pull/559/commits/f2d81eee74bf5fb338aebdec0ed9bb765d321de2#diff-e3147933b179a3ade1d900049f1fa55bL26-R39